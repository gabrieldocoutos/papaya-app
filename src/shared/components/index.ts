import { Text } from "./Text/TextComponent";
import { View, SafeAreaView } from "./View/ViewComponent";

export { Text, View, SafeAreaView };
