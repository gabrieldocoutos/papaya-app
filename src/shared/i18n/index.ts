import homeScreen from "./homeScreen.json";

export { homeScreen };
