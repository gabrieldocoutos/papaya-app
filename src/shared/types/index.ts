import type { RootStackParamList } from "./navigatorTypes";
export { RootStackParamList };
