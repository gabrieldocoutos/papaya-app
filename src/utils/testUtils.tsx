import React, { FC, ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react-native";
import { ThemeProvider } from "styled-components/native";
import { theme } from "shared/configuration";

const AllTheProviders: FC = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, "wrapper">) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react-native";
export { customRender as render };
