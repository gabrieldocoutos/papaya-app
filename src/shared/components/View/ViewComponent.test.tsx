import React from "react";
import { render } from "utils/testUtils";

import { View, SafeAreaView } from "shared/components";

test("snapshot View", () => {
  const { toJSON } = render(<View />);
  expect(toJSON()).toMatchSnapshot();
});

test("snapshot SafeAreaView", () => {
  const { toJSON } = render(<SafeAreaView />);
  expect(toJSON()).toMatchSnapshot();
});
