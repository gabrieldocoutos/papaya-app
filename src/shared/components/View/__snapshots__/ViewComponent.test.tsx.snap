// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot SafeAreaView 1`] = `
<RNCSafeAreaView
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
    ]
  }
/>
`;

exports[`snapshot View 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
    ]
  }
/>
`;
